### PR TITLE
add graph_name to functorch

### PR DESCRIPTION
**Pull request by yushangdi**
### Description
Add utilities to get the graph being compiled for debugging purposes.

used in this PR: https://github.com/pytorch/torchdynamo/pull/665

### CODE REVIEW

File: functorch/functorch/_src/aot_autograd.py [hunk 1]
```diff
@@ -202,14 +220,23 @@ def forward(ctx, *flat_tensor_args):
                             def fake_fn(primals, tangents):
                                 return fx_g(primals, tangents)
                             fx_g = make_fx(functionalize(fake_fn))(*joint_inputs)
+
+                global graph_being_compiled
```
Comment (Chillee) [inline functorch/functorch/_src/aot_autograd.py:224]: I probably prefer a context manager here.

Comment (yushangdi) [reply functorch/functorch/_src/aot_autograd.py]: fixed

**Review by Chillee — APPROVED** [review]
LGTM!

**Pull request closed without merging**